### PR TITLE
feat(audience-session): session lifecycle (SDK-145)

### PR DESCRIPTION
**Pull request by ImmutableJeffrey**
## Summary
- Adds `Runtime/Core/Session.cs` — pure C# session lifecycle emitting `session_start` on Init (or consent upgrade from None), `session_heartbeat` every 60s while foreground, and `session_end` on Shutdown or Resume after a pause longer than 30s. `durationSec` is engagement time only (excludes pause).
- Wires `ImmutableAudience.Init` / `SetConsent` / `Shutdown` / `ResetState` to create, swap, and dispose the active Session under `_initLock`. Routes `OnPause` / `OnResume` from the Unity lifecycle bridge to the session. Reset ends the old session silently and starts a new one, matching Web SDK `reset()`.
- Adds internal `TrackDelegate` for the Session's `_track` callback.
- Hardens `Log.Emit` with try/catch so a throwing Writer cannot escape `SafeTrack` and kill the Timer thread.
- Renames wire field `duration` → `durationSec` on `session_heartbeat` / `session_end` to match the [Unity SDK Event Reference v1](https://immutable.atlassian.net/wiki/spaces/~84417537/pages/3883663502).
- Grows the test suite from 180 to 220 — heartbeat quiescence while paused, extended-pause rollover, clock-rewind clamps, exception containment on `_track` and `PerformanceSnapshotProvider`, double-Start timer drain, concurrent consent-upgrade-from-None stress, concurrent Init+SetConsent consistency, Reset-starts-new-session.

Linear: [SDK-145](https://linear.app/imtbl/issue/SDK-145)

---

> [!NOTE]
> **Medium Risk**
> Adds a new session lifecycle with background heartbeats and refactors `Init`/`SetConsent`/`Reset`/`Shutdown` locking and teardown ordering; mistakes here could cause event leaks, timer/thread issues, or missed/duplicated session events under concurrency.
> 
> **Overview**
> Adds a new `Session` component that emits `session_start`, periodic `session_heartbeat`, and `session_end`, with *engagement-time* `durationSec` (pause-adjusted), heartbeat quiescence while paused, and rollover to a new session after extended pauses.
> 
> Wires session creation/teardown into `ImmutableAudience` so sessions start on `Init` (or consent upgrade from `None`), end on `Shutdown`, and `Reset()` swaps to a fresh session while purging the queued `session_end` for Web-SDK parity; also adds internal `OnPause`/`OnResume` routing and restructures `Shutdown`/`SetConsent`/`ResetState` to release `_initLock` before blocking teardown work.
> 
> Hardens logging (`Log.Emit`) and session tracking/snapshot callbacks to swallow exceptions (especially on timer threads), and expands tests substantially to cover lifecycle ordering, pause/clock edge cases, exception containment, and concurrency stress around consent/session state.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit ebe26c8bb3b9d1b80029201a572873aece4ee130. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: src/Packages/Audience/Runtime/ImmutableAudience.cs [hunk 1]
```diff
@@ -451,105 +495,114 @@ await transport.SendBatchAsync().ConfigureAwait(false))
...
+                if (_transport != null)
                 {
-                    var send = _transport.SendBatchAsync();
-                    if (!send.Wait(timeoutMs))
+                    var timeoutMs = _config?.ShutdownFlushTimeoutMs ?? 2_000;
+                    try
                     {
-                        Log.Warn($"Shutdown flush exceeded {timeoutMs}ms — abandoning. " +
-                                 "Queued events remain on disk and will retry on next startup.");
+                        var send = _transport.SendBatchAsync();
```
Comment (nattb8) [inline src/Packages/Audience/Runtime/ImmutableAudience.cs]: Lots of blocking calls in a lock now, this could block other functions that needs the lock too.
Same in Reset()

Could the lock only clear fields and invalidate states? All other blocking calls be outside? Does that work?

**Review by nattb8 — APPROVED** [review]
